### DISCUSSION TITLE
Welcome to CoastlineExtraction Discussions!

### DISCUSSION THREAD

**Discussion by fwitmer** [fwitmer/CoastlineExtraction#12] [Announcements]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by peeplika** [comment]
Hello, 
I am Priya Yadav, a sophomore pursuing a degree in Computer Science and Engineering. This project seems to suit my interest as I am comfortable with Python and studying machine learning. I have coding experience in areas including data analysis and android development. 
So I'd appreciate any advice on how I should get started.
Thanks

**Comment by Aashish-100** [comment]
Warm Greetings,
I'm Aashish Raheja, a Computer Engineering graduate from India. My expertise lies in using Data Science, Machine Learning, and Deep Learning to tackle climate change issues, particularly in geo-spatial data science for satellite image classification. I'm eager to contribute to the Automated coastline extraction for erosion modeling in Alaska project and would value your insights on where to begin. I've started exploring the DeepWaterMap algorithm and would appreciate your guidance.
Thank you & Best Regards,
Aashish Raheja.

**Comment by Aashish-100** [comment]
Hello @fwitmer, while reviewing the Coastline Extraction repository, I've been familiarizing myself with the function of each file. I have a query regarding obtaining a sample dataset to initiate my work. Could you kindly suggest some data sources for this purpose? Thank you very much!

**Reply by fwitmer** [reply]
Thanks very much to everyone for your interest in this project! To help get you @Aashish-100 get started, I added some sample imagery to the repository. If you would like to explore and download additional PlanetLabs imagery, you can request access here: https://www.planet.com/markets/education-and-research/.

**Comment by Abdulhaadi07** [comment]
Greetings,
My name is Abdul Hadi, a 2022 CSE-Grad. I'm proficient at Python, Java Programming Languages. Also my focus lies around Artificial Intelligence & Machine Learning Models. Developed a few projects in that domain as well. Coming to the project, I have understood the aim and its outcome. As far as my understanding about this project goes, by leveraging advanced deep learning techniques like CNNs and using models such as U-Net or DeepLabv3+, we can try to enhance the accuracy and efficiency of coastline extraction from satellite imagery. I'll give my best in contributing my skills to this project. Might need the guidance of the Mentor to reach the project with efficacy.
Thank You.

**Reply by Abdulhaadi07** [reply]
Hello @fwitmer. Hope you find my message. Would love to hear from you.

**Comment by Aashish-100** [comment]
Thank you very much for the assistance @fwitmer. Will get back to you as I keep making updates on the project.

**Comment by Shashankss1205** [comment]
Greetings,

I'm Shashank Shekhar Singh, a second-year student at IIT BHU, India. My passion lies in harnessing the power of Data Science, Machine Learning, and Deep Learning to address pressing environmental issues like implementing geo-spatial data science for satellite image analysis.

I am eager to contribute to the project on Automated coastline extraction for erosion modeling in Alaska. I would be highly grateful for your guidance on where to start. Currently, I am immersing myself in understanding the algorithms, functions and its applications. 

Thank you sincerely,

Shashank Shekhar Singh

**Comment by Ayush3941** [comment]
Hello everyone ,

I'm Ayush Kumar Gaur, a first-year Btech computer science enthusiast aims to research on the captivating world of AI and ML  to harness its full potential and to apply it in various fields 

i am proficient in python and understanding maths behind machine learning models such as linear regression and decision trees and also have a grasp over python libraries such as numpy, matplotlib and pandas

in order to deepen my understanding in this field i wish to work and contribute in real world projects with an interactive community and understand dynamics of real world problems like Global warming to improve myself

currently i am learning python libraries such as rasterio and rastertools and other libraries that were unknown to me and used in this  project ,to understand to code and contribute to it as well.

Thank You

**Comment by Debarpan08** [comment]
Hey @fwitmer,  I have been fidgeting around with the codebase for a while. I came across the ndwi_labels.py file, figured out what's going on. I was thinking how about trying out the EWI (Enhanced Water Index) or MNDWI(Modified Normalized Difference Water Index) algorithms instead of NDWI. Considering the terrain, EWI might outperform MNDWI at some places. Later on, we might design and train a model architecture capable of accurately segmenting coastlines from satellite imagery. Kindly let me know what you think about it!

**Reply by Debarpan08** [reply]
@fwitmer please let me know what you think about it!

**Reply by fwitmer** [reply]
Thanks for those suggestions. It is certainly worth considering other indices that might help, but the PlanetScope imagery has limited spectral bands, so I don't think the EWI or MNDWI are possible to calculate.

**Comment by MadJokkerr** [comment]
@fwitmer greeting one and all i am Suvrat Patel I have my Bachelors Degree In Artificial Intelligence And Machine Learning Currently I am Writing my Research paper as well as i have experience in implementing research paper project to real go, i have worked on similar projects with similar skill set i have skills on python, Machine Learning, Deep learning, Geospatial data Processing as well as Satellite imagery, in past i have worked on similar project with ISRO(Indian Space Research Organization ) Satellite imagery which is real time based project for Human Lives.

**Comment by RuoyuLi97** [comment]
Greetings to @fwitmer and all, my name is Ruoyu Li and I am an incoming grad student of Information Systems at Northeastern University in USA this fall. I am very passionate to work in the coastline extraction project. 

I am proficient in python and machine learning. I also have experiences in geospatial data processing. Meanwhile, as a former landscape architect, I care a lot about the ecosystem and I completed many related projects both in design and coding(such coral reef resistency design against the climate change, water and terrain analysis with python etc). Among those projects, they all have research involvement as I really enjoy the combination of research and practices. 

I really want to contribute to this project as it matches my professional pursuit - to raise more social awareness about ecosystem via technology. I am currently going over the documentation and get familiar with the files.\

Thank you
Ruoyu Li

**Comment by staru09** [comment]
Greetings
I have read this paper and they have used different types of U nets for a task that we are doing and they have pretty good accuracy. Can we follow a similar approach?

**Reply by fwitmer** [reply]
Yes, using a U-Net CNN and/or other similar methods makes sense.

**Comment by FaizalJnu** [comment]
Hello @fwitmer Sir!

I am a junior in a Computer Engineering course here at JNU, New Delhi. I feel quite strongly for active climate control and my efforts to reverse the or contain the damage in any form. I've drawn inspiration to work in this domain from a recent talk by a Yale Climate Fellow in our college. 

Point is my motivations to work on this project remain crystal clear, as for the project needs - As I understand you have already done the following:
1. Expanding the training data in the field of including manual annotations by experts, crowdsourced data or synthetic data generation techniques?
2. Incorporating multi sensor data?
3. Incorporating ancillary data?

Also, I’d like to highlight my previous work in the ML/AI, where I’ve done considerable work in a number of different domains, including: Reinforcement learning, NLP-LLMs(current) and Supervised ML Projects. 

After starting with the code base review, I came across your preprocessing methods and would like to add, if we implemented Elastic Deforming, Color jittering and Random Rotating, we'd be closer to achieving our end goal of a more clear and precise model. I've so far been trying to implement this on EuroSat dataset to confirm my theory. I've also sent a request for the original planetLabs dataset and as soon as I get it, I'll try to generate a PR so you can check the work done.

**Comment by zz39** [comment]
Hi 👋 @fwitmer and everyone interested in the Coastline Extraction project!

I'm Wayne Zhou, a Master's student in Computer Science at Northeastern University. Previously, I worked as an architectural designer specializing in building information modeling (BIM), GIS, and green technology. I'm passionate about projects with real-world impact and problem-solving potential. Outside of coding, I enjoy classical arts, modern furniture, and exploring nature.

I've been reviewing the repository and have a few questions to get started:

1. Readme and Dependencies: The project seems to lack a comprehensive README file and clear instructions for setting up dependencies. Is there a requirements.txt file listing all necessary Python packages?
2. Accuracy Evaluation: How is the accuracy of the contour line extraction currently measured? Additionally, what's the performance of the DeepWaterMap algorithm without the NDWI thresholding method?
3. Future Coastline Erosion Prediction: While the project focuses on extracting contour lines from imagery, I was wondering if integrating additional factors into a predictive ML model might be beneficial. These could include sea level, annual wave energy, temperature, etc. This model could then predict future coastline erosion and be verified against historical data. (I understand this might be outside the current project scope, but I just wanted to share the idea:D)

Thanks in advance for your time and guidance!

**Comment by sonnetx** [comment]
Hello everyone!
I'm Sonnet Xu, a current freshman studying computer science at Stanford University, with a strong interest in applying machine learning to environmental science research. I've done quite a bit of work with Google Earth Engine and remote sensing, specifically satellite imagery, in the past so I'm excited to help tackle the problem of Coastline Extraction.

Looking forward to learning more about the project and exploring how I can contribute!

**Comment by srinivaspavan9** [comment]
Hi, 
I have a few questions with regards to your approach and the dataset.
Can i know what all spectral indices you are using currently for the model ? I have seen that you are using the normalized difference water index i would like to know if there are any other indeces you are using apart from that.
For the dataset, will more bands be available for use? i see  in the data_preprocessing file, only up to 5 bands are being accounted for

**Comment by rawann31** [comment]


**Reply by pradeeban** [reply]
Hi @rawann-elframawy,
You can email your proposal draft to the mentor for feedback. But the deadline is fast approaching on April 2. So, even if you do not hear back with any reviews on your proposal, make sure to submit on GSoC site on time!

We will continue our reviews after the submission deadline.

**Reply by rawann31** [reply]
Thanks so much I am writing it now I want to ask if the proposal has max words that we should exceed ?

**Reply by pradeeban** [reply]
There is no max words. Make sure to follow the guidelines - https://github.com/uaanchorage/GSoC/blob/main/CONTRIBUTOR-GUIDE.md

**Reply by rawann31** [reply]
Can I reorder sections or I should follow the sections order of the proposal in the guide ? I mean follow my own order but same section.

**Reply by pradeeban** [reply]
I don't see a rationale for reordering the sections.

But as the document says, "The contributors are encouraged to follow this template. However, they are not expected to follow this template strictly. They are advised to include all the requested information in their application."

**Comment by varun-aahil** [comment]
Hi @fwitmer, @pradeeban, and everyone,

I’m Varun, a CS undergrad, and I came across this project through GSoC 2025 and found the idea of using satellite imagery for coastal erosion analysis really interesting! I’ve been exploring NDWI visualization and would love to contribute, possibly by improving error handling for missing/invalid tiles.

Is this forum still active? Also, does the project have a Slack or another communication platform for discussions?

Looking forward to contributing!

Thanks!

**Reply by varun-aahil** [reply]
@rawann31 Just following up on my previous message. Can you please assist me with this?

Looking forward to hearing from you!

Thanks!

**Reply by pradeeban** [reply]
"Is this forum still active? Also, does the project have a Slack or another communication platform for discussions?"

Yes, this is active. We do not have Slack or other forums - to keep the conversations open and visible. (Our GitHub discussions are specified as the preferred communication channel).
Usually mentors do not respond if there is no question (as there are several introduction messages with no question in them). If you have specific questions, ask right away!

**Reply by varun-aahil** [reply]
@pradeeban Thanks for the response! I appreciate the clarity on keeping discussions open here.

I was looking into potential ways to contribute and was thinking to start by improving error handling for missing or invalid tiles. Would this be a meaningful contribution? Do missing tiles currently cause issues in the coastline extraction pipeline, or is it something that’s already well-handled?

If there are any specific areas where contributions are most needed, I’d love to get started on them!

Looking forward to your thoughts.

**Reply by pradeeban** [reply]
@varun-aahil, Mentors (@fwitmer and @rawann31) for the project will have a better answer for your question since I am not familiar with this project enough.  

But as org admin, I could give you some general suggestions (which are actually true for all the organizations, not just Alaska):
You could always open an issue in GitHub and create a pull request (if you can and if your time permits).
In the best case, the pull request will be merged and you will be positively evaluated as that establishes your experience and expertise.
In the worst case, your issue and pull request may be closed without merge. Even in that case, you can (and should!) still mention your issue and pull request links in your GSoC proposal, as that will highlight your effort.

**Reply by varun-aahil** [reply]
Thanks, @pradeeban! I really appreciate the advice and will definitely keep that in mind while contributing.

@fwitmer, @rawann31, since there aren’t any open issues, I’ll start exploring the codebase and documentation to better understand the extraction pipeline. If I find areas that could use improvement, I’ll open an issue or a PR.

I’ll reach out if I have any questions along the way.

**Reply by varun-aahil** [reply]
Hi @rawann31,

I hope you're doing well. Just wanted to follow up on the pull request I submitted addressing the inconsistent date format issue in `label_inputs.py`. When you have a moment, could you please review the changes and provide your feedback?

Additionally, I plan to raise more PRs as I continue exploring and improving the codebase. It would be great if you could check them from time to time.

Thank you!

**Reply by varun-aahil** [reply]
@rawann31 Can I email you my proposal, if yes can you please check it out

**Comment by Asta1984** [comment]
> 
> ## 👋 Welcome!
>   We’re using Discussions as a place to connect with other members of our community. We hope that you:
>   * Ask questions you’re wondering about.
>   * Share ideas.
>   * Engage with other community members.
>   * Welcome others and are open-minded. Remember that this is a community we
>   build together 💪.
> 
>   To get started, comment below with an introduction of yourself and tell us about what you do with this community.
> 
> 
>

**Comment by rushichavda** [comment]
Hello Everyone!
I am Rushi, final year undergrad student in IIT Bombay. I have worked on the geospatial data and geospatial foundational model in my internship at IBM Research. Would love to contribute!

**Reply by rawann31** [reply]
Hello @rushichavda 

Thanks for your interest in the project! If you’d like to contribute, you’ll need to submit a strong GSoC application that highlights your experience with geospatial data, deep learning, and frameworks like PyTorch or TensorFlow.
If you have any questions, feel free to reach out!

**Reply by rushichavda** [reply]
I have worked on raster data, from google earth engine, with several processing libraries. i have also used geospatial foundational model prithvi-100 - i  have finetuned it. i have developed pipeline to process the satellite images.

**Reply by rushichavda** [reply]
would this experience help? this i did in my internship at IBM research.

**Comment by protocorn** [comment]
Hey everyone, Sahil Chordia here! 👋


Really excited about the possibility of participating in GSoC 2025, and I'm super interested in projects that leverage AI, machine learning, and automation. 

I am a student at the University of Maryland, pursuing a Master's in Data Science, I'm eager to explore how we can apply these technologies to enhance coastline extraction methods, maybe through more efficient data processing or automated analysis. I'm really looking forward to hearing your thoughts, learning from all of you, and brainstorming some ideas!

Cheers,
Sahil

**Comment by PrasannaPal21** [comment]
Hey everyone, I'm Prasanna.

I’ve been going through the CoastlineExtraction repo lately and got really interested in the Planet imagery + NDWI workflow. I’m hoping to understand the pipeline better and start helping out, especially around improving the segmentation side of things.

Looking forward to learning from you all and contributing where I can.

**Comment by lakshyadchad** [comment]
Hi mentors (@fwitmer),

My name is Lakshya, and I am a student developer interested in contributing to the Automated Coastline Extraction project.

I have a background in Python and I am currently ramping up on the geospatial stack (rasterio, geopandas). I have successfully set up the local environment and have written a simulation script to replicate the NDWI (Normalized Difference Water Index) logic mentioned in the project overview to understand how the water masking works.

You can see my NDWI reproduction script here: https://github.com/lakshyadchad/alaska-gsoc-prep

I am now moving on to testing the pipeline with the sample PlanetLabs imagery provided in the repo. I am particularly interested in the "Cliff Area Segmentation" challenge and will be looking for guidance on where I can be most helpful with testing or documentation in these early stages.

Looking forward to contributing!

**Reply by fwitmer** [reply]
@lakshyadchad  Thanks for your interest, Lakshya! We just updated the master branch, so please be sure to take a look at the latest version of the code base.

**Reply by lakshyadchad** [reply]
Okay @fwitmer sir, I will look at them. Thank you for updating me.

**Reply by lakshyadchad** [reply]
Hi @fwitmer sir and mentors,
I want to give you an update on my current progress. I am trying to expand the dataset with more recent imagery (2020-2025), but I am facing an issue obtaining a Planet API key to gather these resources. I have already applied for one, but the approval process might take some time.

Do you have any suggestions on how to proceed or alternative options to look for?
Looking forward to your feedback.

**Comment by PrasannaPal21** [comment]
Hello @fwitmer ,

I'm Prasanna, interested in GSoC 2026. I talked with @Ritika-K7  about a month ago and she suggested I go through the codebase and get familiar with it. I've been exploring the dev branch and tried running the pipeline locally.

I noticed the project doesn't have automated tests yet. I think adding testing infrastructure would be useful - it makes it easier to verify changes work and helps future contributors feel confident their code won't break things.

So I've started building a testing infrastructure for the project. I've set up pytest framework with basic fixtures and added 9 tests for config loading (all passing). 
My plan is to continue adding tests for:
- CRS validation
- NDWI calculations  
- Main preprocessing steps

Branch: https://github.com/PrasannaPal21/CoastlineExtraction-gsoc/tree/testing-infra

Would appreciate any feedback - is this useful? Any specific parts you'd want tested first?

**Reply by PrasannaPal21** [reply]
UPDATE - 
added 9 CRS validation tests. Tests check CRS detection for rasters/vectors and matching between files. All passing.
why it matters - CRS mismatches are a common source of bugs in geospatial processing - images and shapefiles need to be in the same coordinate system or the analysis fails. These tests catch those issues early.

Progress: pytest setup , config tests , CRS tests 
Next: NDWI calculation tests

https://github.com/PrasannaPal21/CoastlineExtraction-gsoc/tree/testing-infra

**Reply by PrasannaPal21** [reply]
Quick Update: Just added 16 tests for NDWI calculations and raster tools. They check everything from sliding window setup and noise cleanup (via morphological transforms), to k-means clustering, the actual NDWI math, and Otsu thresholding for clean water/land splits.

Why i added these: NDWI is the heart of our water/coastline detection from satellite images - positive values flag water, negatives mean land. Passing these tests means no math bugs that could mess up coastlines later.

Progress so far: pytest basics, config checks, CRS handling, and NDWI.
Next: Spatial utility tests.

Branch: [github.com/PrasannaPal21/CoastlineExtraction-gsoc/tree/testing-infra](https://github.com/PrasannaPal21/CoastlineExtraction-gsoc/tree/testing-infra)

**Reply by PrasannaPal21** [reply]
hello
added 14 tests for spatial analysis & GIS utils, covering log_spatial_info, bounding box overlaps, geometry ops (intersections, clipping, point checks), and file paths

this ensures satellite images actually overlap with shapefiles before heavy processing so no wasted compute on misaligned data.

Progress: pytest, config, CRS, NDWI, spatial utils.
Next: Integration tests.

branch - https://github.com/PrasannaPal21/CoastlineExtraction-gsoc/tree/testing-infra

**Reply by PrasannaPal21** [reply]
Hi @fwitmer sir,

I added tests for NDWI/raster stuff, spatial checks like overlaps and geometry, and some integration tests that run the full pipeline on real Planet satellite images. I tried these on a sample input, all the tests were passing. the tests I added were from my knowledge of testing, would like to know your view on them if they make sense in the project or not.

This means we catch math bugs, data mismatches, and workflow glitches early, super important for reliable coastline extraction without wasting time on bad data.

things covered: setup, config, CRS, NDWI, spatial utils, integrations, docs, and requirements. 

checkout the testing module here - https://github.com/PrasannaPal21/CoastlineExtraction-gsoc/tree/testing-infra

@fwitmer, @rawann31, @Ritika-K7 I would appreciate your views on the testing infrastructure for the project.

**Comment by Jiya873** [comment]
Hi @fwitmer,
I'm Jiya Gupta, a pre-final year student at IIT Kharagpur. I'm very excited to express my interest in the "Automated coastline extraction for erosion modeling in Alaska" project for GSOC 2026. I wanted to share that I have set up the project locally successfully reproduced the initial extraction results. I noticed a few minor issues in the README.md and I opened an [issue](https://github.com/fwitmer/CoastlineExtraction/issues/78) as well. Currently I am exploring project infrastructure and would like to contribute to this project. 
Looking forward for your guidance!

**Comment by fwitmer** [comment]
Jiya, thanks so much for bringing this to our attention. I just requested that LJitendra05 modify their PR which tried to fix this issue.  The dev branch does exist, so hopefully you are able to access it here:

https://github.com/fwitmer/CoastlineExtraction/tree/dev

**Comment by SamtaSakha** [comment]
Hi @fwitmer ,

I'm Samta Sakha, a  Data Science and AI student from India, and I'm very interested in the  **Automated Coastline Extraction for Erosion Modeling**. I have been exploring the repository and understanding the workflow around NDWI-based labeling and the DeepWaterMap training pipeline. While reviewing the preprocessing steps, I noticed that Planet imagery provides a Usable Data Mask (UDM) that identifies clouds, shadows, and other low-quality pixels. Integrating the UDM mask during preprocessing could help filter unusable regions, improve the quality of NDWI-generated labels, and reduce noise in the training dataset. I opened an **issue** [#89 ](https://github.com/fwitmer/CoastlineExtraction/issues/89) regarding this and would be interested in exploring how UDM masking could be incorporated into the preprocessing pipeline.

Please let me know if this would be a useful direction to explore, and I would be happy to contribute to its implementation.

**Comment by MennaEzzelarab** [comment]
Hello,


I am Menna Ezz Elarab, a senior year Computer Engineering student from Ain Shams University in Egypt. I have skills in Python, machine learning, and deep learning. I haven't contributed to any open source projects before, but I am willing to take on a new challenge this year.

This summer, I am interested in contributing to Project Idea 1: **Automated coastline extraction for erosion modelling in Alaska**, as it aligns with my previous experience with a water segmentation project. It was my first experience with working on multi-band satellite imagery and geospatial data processing. You can find this project here: (https://github.com/MennaEzzelarab/water_segmentation/tree/main).

**Comment by modi02** [comment]
Hello @fwitmer,
I'm Raj Modi, and I'm a computer science student from India. I found this project through GSoC 2026, and the problem really interested me. Coastal erosion is affecting real communities in Alaska, and I want to work on projects like this that use satellite imagery to track it.
I haven't been doing geospatial stuff for very long, but I've been using Python and ML for a while. I just started looking through the repo and trying to figure out how the pipeline works. I'm still trying to figure out how the NDWI thresholding fits into the creation of training data, but I'm getting there.
I'm excited to learn from everyone here and hopefully add something useful.

**Comment by fwitmer** [comment]
Thanks for everyone's interest in our project! We are excited to start reviewing applications. If you are still wondering where to focus your efforts, you might think about how you can improve the automatic generation of training/labeled data. That is what we are trying to do with the NDWI sliding window, but it needs improvement. Once we have a robust set of training data, then the various models should do much better. Suggesting improvements/ideas on the modeling side are also welcome. Good luck everyone!

**Comment by romit-basak** [comment]
Hi everyone,

I'm Romit Basak, a first-year MS Computer Science student at Northeastern University (Khoury College), working in GeoAI and remote sensing ML. I'm interested in applying to Project 1 (Automated Coastline Extraction) for GSoC 2026.

My relevant experience comes from developing a satellite-based water detection and flood prediction system for urban environments. I've worked extensively with multi-band satellite imagery including Sentinel-1 SAR and optical sensors, NDWI/MNDWI thresholding for water body detection, and pseudo-label generation using optical imagery to supervise SAR-based models — a workflow closely related to the NDWI-with-majority-sliding-window approach described in this project. I've also worked through the specific failure modes of threshold-based water detection in challenging environments: false positives from radar shadow in dense urban SAR, building-shadow contamination in optical NDWI, and the class imbalance problem in rare-event detection.

I looked through the CoastlineExtraction repository and have a few questions:

1. The DeepWaterMap retraining on PlanetLabs imagery with NDWI pseudo-labels is the core open task — is there a specific band combination from PlanetLabs (RGB + NIR, or including the UDM mask channels) that prior work has found most useful for the NDWI labeling step?
2. For the cliff area segmentation improvement, is the current bottleneck in the training data (too few cliff examples), the model architecture, or both?
3. Is SWIR data from PlanetLabs' SuperDove already in the pipeline, or would SWIR integration be a fresh addition for this GSoC cycle?

Happy to take on any starter tasks. I'm familiar with the geospatial Python stack (rasterio, GDAL, GEE, PyTorch) and have experience with U-Net-style segmentation on satellite imagery patches.

Looking forward to the conversation.

Romit Basak

**Comment by MennaEzzelarab** [comment]
Hello @fwitmer and @Ritika-K7,

I am currently finishing my proposal for GSoC 2026. I have a few questions.

1. Is the community bonding period only for getting to know mentors and understanding the project, or should I include development progress to be done in that time?
2. How do I get feedback on my draft proposal? 

Thanks in advance.

**Reply by pradeeban** [reply]
1. Yes, it is only to understand the project better, and perhaps refine your project design. Coding/development happens during the coding period.
2. Might be actually a bit late to get feedback as mentors have a huge backlog now with several proposals to review and the deadline being March 31st. Please upload your proposal to GSoC Portal on time and the mentors will contact you before April 21st if they have questions or comments. 
     

March 31 - 18:00 UTC

    GSoC contributor application deadline

April 21 - 18:00 UTC

    GSoC contributor proposal rankings due from Org Admins

**Reply by MennaEzzelarab** [reply]
okay no problem. Thank you

**Reply by MennaEzzelarab** [reply]
@pradeeban If I choose to extend the time and add 2 additional months, should I ignore the midterm and final evaluation dates mentioned in the GSoC program timeline, or should I add them to my timeline as they are?

**Reply by pradeeban** [reply]
Then you should add the updated timeline in your proposal. The mid-evaluation will come right in the middle as your extended timeline. So, if you extend it by 8 weeks, your mid-evaluation comes 4 weeks later than the original mid-evaluation time.

**Reply by MennaEzzelarab** [reply]
Thanks for your help.
